### PR TITLE
refactor!: Hard-code vertex fitter, finder + density combinations

### PR DESCRIPTION
**Pull request by paulgessinger**
They're not actually exchangeable, there's a 1:1 correspondence that they can be used in. Therefore, we don't need to template the finders on the only respective valid density.

- `AdaptiveGridDensityVertexFinder` => `AdaptiveGridTrackDensity`
- `AdaptiveMultiVertexFinder` => `AdaptiveMultiVertexFitter`
- `TrackDensityVertexFinder` => `GaussianTrackDensity`
- `IterativeVertexFinder` => `FullBilloirVertexFitter`

The main reason for this is that the vertex fitters and the track densities do not actually have common interfaces. This is probably desirable, but out of scope for this refactor, and should probably be implemented with an interface like `IVertexFinder`.

Part of:
- #2842 

Blocked by:
- #2951

### CODE REVIEW

**Comment by paulgessinger** [comment]
@andiwand this is ready now!

File: Core/include/Acts/Vertexing/AdaptiveMultiVertexFinder.hpp [hunk 1]
```diff
@@ -25,19 +26,9 @@
 
 namespace Acts {
 /// @brief Implements an iterative vertex finder
-///
-////////////////////////////////////////////////////////////
-///
-/// Brief description of the algorithm implemented:
-/// TODO
-///
-////////////////////////////////////////////////////////////
```
Comment (andiwand) [inline Core/include/Acts/Vertexing/AdaptiveMultiVertexFinder.hpp:34]: 😄

**Review by andiwand — APPROVED** [review]
🥰

**Comment by acts-project-service** [comment]
## :red_circle: Athena integration test results [8ccefe18f1263706567ba5848e1ed23ee520fb72]
### Build job with this PR failed!
Please investigate the [build job](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/36310656) for the [pipeline](https://gitlab.cern.ch/acts/acts-athena-ci/-/pipelines/6930034)!

**Pull request merged**